src/main/scala/djinni/generator.scala

```diff
@@ -716,26 +759,28 @@ abstract class Generator(spec: Spec) {
   def writeEnumOptionNone(
       w: IndentWriter,
       e: Enum,
-      ident: IdentConverter
+      ident: IdentConverter,
+      delim: String = "="
   ): Unit = {
-    for (
-      o <- e.options.find(_.specialFlag.contains(Enum.SpecialFlag.NoFlags))
-    ) {
+    for (o <- e.options.find(_.specialFlag == Some(Enum.SpecialFlag.NoFlags))) {
       writeDoc(w, o.doc)
-      w.wl(ident(o.ident.name) + " = 0,")
+      w.wl(ident(o.ident.name) + s" $delim 0,")
     }
   }
 
   def writeEnumOptions(
       w: IndentWriter,
       e: Enum,
-      ident: IdentConverter
+      ident: IdentConverter,
+      delim: String = "=",
+      optionSuffix: String = ""
   ): Unit = {
     var shift = 0
     for (o <- normalEnumOptions(e)) {
       writeDoc(w, o.doc)
       w.wl(
-        ident(o.ident.name) + (if (e.flags) s" = 1u << $shift" else "") + ","
+        ident(o.ident.name) + (if (e.flags) s" $delim 1$optionSuffix << $shift"
```

Comment:
A bit hideous, but... 🤷‍♂️

Reply:
Do we want to be super configurable on that level? 
It adds complexity, and complexity kills projects. And confuses users.
I do not see any return value for adding this option, but I am open for arguments

If possible, and there is no solid reason for the change, like a bug, I would like to keep the expected files in testing as they are.

Reply:
Ahh, this option is to restore the former behaviour so that non-wasm test file expectations are unchanged. How best to achieve this - by overriding behaviour in generator subclasses, or by making this method take more arguments to be more configurable - it's not clear to me. I think I would do things very differently if I were starting from scratch, but yeah, this is not that.

Reply:
Does wasm not support the current generated values of `1u << 4` ?

Reply:
Same question here, can't wasm have the same declaration with flag + u values and enums with no values as the pure C++ part?

Reply:
wasm is like :

```
    EM_JS(void, djinni_init__my_enum_consts, (), {
        Module.MyEnum =  {
            /** enum option comment */
            OPTION1 : 0,
            OPTION2 : 1,
            OPTION3 : 2,
        }
    })
```

Reply:
I should probably post this instead -

`
    EM_JS(void, djinni_init__my_flags_consts, (), {
        Module.MyFlags =  {
            NO_FLAGS : 0,
            /** flag option comment */
            FLAG1 : 1 << 0,
            FLAG2 : 1 << 1,
            FLAG3 : 1 << 2,
            ALL_FLAGS : 0 | (1 << 0) | (1 << 1) | (1 << 2),
        }
    })
`

So no unsigned suffix, and I haven't checked, but I assume the "ALL_FLAGS" option cannot take the OR of the other flags because they are not yet set.

Reply:
That might mean we must figure out in what generation mode we are and select the generation accordingly.
 Or there is another way ... but that would probably mean more re-write.
That sound like the `$optionSuffix` is the way to go, or do I misunderstand?

Reply:
for EM_JS, the content in {} is evaluated as javascript, so anything you can do In JS is valid.

Reply:
I think the $optionSuffix approach is simplest, because it only involves editing `generator.scala`. Otherwise I think we'd have to make a common base class for the generators that have unsigned flags, or maybe a mixin? But in the end I think the $optionSuffix makes sense here.

Reply:
probably a simpler option would be to pass a closure to allow the caller to formulate the value as needed.

something like 
```
def writeEnumOptions( w: IndentWriter, e: Enum, ident: IdentConverter, flag: ( Enum.Option, Int ) => String ) {
  var shift = 0
     for (o <- normalEnumOptions(e)) {
       writeDoc(w, o.doc)
       w.wl(
         ident(o.ident.name) + (if (e.flags) flag( o, shift ) else "") + ","
       )
       shift += 1
}  
```

then the default can be 
```
def writeEnumOptions( w: IndentWriter, e: Enum, ident: IdentConverter, flag: ( Enum.Option, Int ) => String ) {
    writeEnumOptions( w, e, ident, ( o: Enum.Option, shift: Int) => s" = 1u << $shift" );
}
```

and any generator can choose use a different flag value generation

Reply:
> Otherwise I think we'd have to make a common base class for the generators

Any generator can provide/use an alternate routine for generating its content it doesn't have to use the common routines in generator.

This might be preferable as to not complicate the common case for a single outlier that needs a specific alternate syntax.

Which I think hands down would be the simpler approach.

Reply:
It looks like there would also need be changes to the writeEnumOptionsNone